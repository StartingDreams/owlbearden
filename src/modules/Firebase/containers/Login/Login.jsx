import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import FontIcon from 'material-ui/FontIcon';
import { Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import firebase from 'firebase';

const auth = firebase.auth;
const provider = new auth.GoogleAuthProvider();

export const loginClick = () => {
  auth().signInWithRedirect(provider);
};

export class Login extends React.Component {
  redirect = () => (<Redirect to={{ pathname: '/' }} />);

  renderLogin = () => (<RaisedButton
    icon={<FontIcon className="material-icons">account_circle</FontIcon>}
    label="Login"
    secondary
    fullWidth
    onClick={loginClick}
  />);

  render = () => (this.props.isAuthenticated ? (this.redirect()) : (this.renderLogin()));
}

Login.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

export const mapStateToProps = state => ({
  isAuthenticated: state.auth.get('isAuthenticated'),
});

export default withRouter(connect(
  mapStateToProps,
)(Login));
