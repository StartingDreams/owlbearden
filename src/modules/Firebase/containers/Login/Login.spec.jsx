import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import { Login, loginClick, mapStateToProps } from './Login';

jest.mock('firebase', () => {
  const signInWithRedirect = jest.fn();
  const authReturn = {
    signInWithRedirect,
  };
  const auth = () => (authReturn);

  auth.GoogleAuthProvider = function Provider() { return { test: 'test' }; };
  return { auth };
});
const firebase = require('firebase');

const auth = firebase.auth;

describe('<Login />', () => {
  it('login click should call signInWithRedirect', () => {
    loginClick();
    // wrapper.find('RaisedButton').simulate('touchTap');
    expect(auth().signInWithRedirect).toHaveBeenCalledWith({ test: 'test' });
  });
  it('mapStateToProps - isAuthenticated should be passed in', () => {
    const state = {
      auth: fromJS({ isAuthenticated: true }),
    };
    const props = mapStateToProps(state);
    expect(props.isAuthenticated).toBe(true);
  });
  it('unauthenticated should match the login snapshot', () => {
    const isAuthenticated = fromJS(false);
    const wrapper = shallow(<Login isAuthenticated={isAuthenticated} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('authenticated should match the home snapshot', () => {
    const isAuthenticated = fromJS(true);
    const wrapper = shallow(<Login isAuthenticated={isAuthenticated} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

