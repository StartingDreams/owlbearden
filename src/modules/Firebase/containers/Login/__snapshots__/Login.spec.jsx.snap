// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> authenticated should match the home snapshot 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`<Login /> unauthenticated should match the login snapshot 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={true}
  icon={
    <FontIcon
      className="material-icons"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      account_circle
    </FontIcon>
  }
  label="Login"
  labelPosition="after"
  onClick={[Function]}
  primary={false}
  secondary={true}
/>
`;
